plugins/woocommerce-beta-tester/src/remote-logging/index.tsx

```diff
@@ -0,0 +1,323 @@
+		try {
+			const response: RemoteLoggingStatus = await apiFetch( {
+				path: `${ API_NAMESPACE }/remote-logging/toggle`,
+				method: 'POST',
+				data: { enable: ! isRemoteLoggingEnabled },
+			} );
+			setIsRemoteLoggingEnabled( response.isEnabled );
+
+			window.wcSettings.isRemoteLoggingEnabled = response.isEnabled;
+		} catch ( error ) {
+			setNotice( {
+				status: 'error',
+				message: `Failed to update remote logging status. ${ JSON.stringify(
+					error
+				) }`,
+			} );
+		}
+
+		if ( window.wcSettings.isRemoteLoggingEnabled ) {
+			initRemoteLogging( {
+				errorRateLimitMs: 60000, // 1 minute
+			} );
+		}
+	};
+
+	const simulatePhpException = async ( context: 'core' | 'beta-tester' ) => {
+		try {
+			await dispatch( OPTIONS_STORE_NAME ).saveOption(
+				'wc_beta_tester_simulate_woocommerce_php_error',
+				context
+			);
+			setNotice( {
+				status: 'success',
+				message: `Please refresh your browser to trigger the PHP exception in ${ context } context.`,
+			} );
+		} catch ( error ) {
+			setNotice( {
+				status: 'error',
+				message: `Failed to trigger PHP exception test in ${ context } context. ${ JSON.stringify(
+					error
+				) }`,
+			} );
+		}
+	};
+
+	const logPhpEvent = async () => {
+		try {
+			await apiFetch( {
+				path: `${ API_NAMESPACE }/remote-logging/log-event`,
+				method: 'POST',
+			} );
+			setNotice( {
+				status: 'success',
+				message: 'Remote event logged successfully.',
+			} );
+		} catch ( error ) {
+			setNotice( {
+				status: 'error',
+				message: `Failed to log remote event.`,
+			} );
+		}
+	};
+
+	const resetPhpRateLimit = async () => {
+		try {
+			await apiFetch( {
+				path: `${ API_NAMESPACE }/remote-logging/reset-rate-limit`,
+				method: 'POST',
+			} );
+			setNotice( {
+				status: 'success',
+				message: 'PHP rate limit reset successfully.',
+			} );
+		} catch ( error ) {
+			setNotice( {
+				status: 'error',
+				message: `Failed to reset PHP rate limit. ${ JSON.stringify(
+					error
+				) }`,
+			} );
+		}
+	};
+
+	const simulateException = async ( context: 'core' | 'beta-tester' ) => {
+		try {
+			await dispatch( OPTIONS_STORE_NAME ).saveOption(
+				'wc_beta_tester_simulate_woocommerce_js_error',
+				context
+			);
+
+			if ( context === 'core' ) {
+				setNotice( {
+					status: 'success',
+					message: `Please go to WooCommerce pages to trigger the JS exception in woocommerce context.`,
+				} );
+			} else {
+				setNotice( {
+					status: 'success',
+					message:
+						'Please refresh your browser to trigger the JS exception in woocommerce beta tester context.',
+				} );
+			}
+		} catch ( error ) {
+			setNotice( {
+				status: 'error',
+				message: `Failed to set up JS exception test`,
+			} );
+		}
+	};
+
+	const logJsEvent = async () => {
+		try {
+			const result = await log(
+				'info',
+				'Test JS event from WooCommerce Beta Tester',
+				{
+					extra: {
+						source: 'wc-beta-tester',
+					},
+				}
+			);
+
+			if ( ! result ) {
```

Comment:
@chihsuan Are you getting a TS error here in your IDE? Its preventing me from building a ip, but I see all possible values from `log` return a Promise that results in a `bool`, so not too sure whats going on. Were you able to build this?

> An expression of type 'void' cannot be tested for truthiness.

Reply:
@psealock I was able to build it. Can you try to rebuild the remote-logging package and see if that fixes it? 🤔  I just released a new version of beta tester via CI so it should able to build.

Reply:
Running `pnpm --filter='@woocommerce/remote-logging' build ` beforehand worked, which means a dependency isn't working as it should. I think it pnpm isn't configured to build all dependencies but the `start` command works as expected using `wp-scripts`.

Fix here: https://github.com/woocommerce/woocommerce/pull/50508